src/main/cpp/subsystems/SwerveDrive.cpp

```diff
@@ -382,10 +398,36 @@ bool SwerveDrive::AutoBalance() {
         OKC_CALL(this->left_back_module_->GetSteerOutput(&this->interface_->left_back_steer_motor_output));
         OKC_CALL(this->right_front_module_->GetSteerOutput(&this->interface_->right_front_steer_motor_output));
         OKC_CALL(this->right_back_module_->GetSteerOutput(&this->interface_->right_back_steer_motor_output));
-    }
 
-    last_yaw_ = this->interface_->imu_pitch;
+
+        // and then if we're balanced, go ahead and stop us
+        if (abs(interface_->imu_pitch) < 1.0) {
+            if (probably_balanced_) {
+                balance_state_ = FINISHED;
+            }
+
+            probably_balanced_ = true;
```

Comment:
I would rework this logic - currently this only gives you one "frame" where the pitch needs to be less than 1 degree. It would be far better to parameterize a timeout, and then only move to the finished state if you are at less than 1 degree for that timeout